src/azure-cli/azure/cli/command_modules/serviceconnector/_credential_free.py

```diff
@@ -338,20 +336,20 @@ def set_target_firewall(self, add_new_rule, ip_name):
             return False
 
     def create_aad_user_in_sql(self, connection_args, query_list):
-        import pkg_resources
-        installed_packages = pkg_resources.working_set
-        # pylint: disable=not-an-iterable
-        psy_installed = any(('pyodbc') in d.key.lower()
-                            for d in installed_packages)
-
-        if not psy_installed:
-            import pip
-            pip.main(['install', 'pyodbc'])
-            logger.warning(
-                "Please manually install odbc 18 for SQL server, reference: https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16 "
-                "and run 'pip install pyodbc'")
+
+        if not is_packaged_installed('pyodbc'):
+            _run_pip(["install", "pyodbc"])
+
         # pylint: disable=import-error, c-extension-no-member
-        import pyodbc
+        try:
+            import pyodbc
+        except ModuleNotFoundError:
+            raise CLIError(
+                "Dependency pyodbc can't be installed, please install it manually with `" + sys.executable + " -m pip install pyodbc`.")
+        drivers = [x for x in pyodbc.drivers() if x == 'ODBC Driver 18 for SQL Server']
+        if not drivers:
+            raise CLIError(
```

Comment:
CLIError is not recommended, please follow this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
Thanks @kairu-ms, it's updated now.